### PR TITLE
 RavenDB-9012 Fix a race condition in replication tests

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: src/Raven.Client/Documents/Replication/ReplicationNode.cs [hunk 1]
```diff
@@ -42,10 +42,10 @@ public string Url
 
         public bool Equals(ReplicationNode other) => IsEqualTo(other);
 
-        public bool IsEqualTo(ReplicationNode other)
+        public virtual bool IsEqualTo(ReplicationNode other)
         {
-            return string.Equals(Url, other.Url, StringComparison.OrdinalIgnoreCase) &&
-                   string.Equals(Database, other.Database, StringComparison.OrdinalIgnoreCase);
+            return string.Equals(Database, other.Database, StringComparison.OrdinalIgnoreCase) &&
```
Comment (ayende) [inline src/Raven.Client/Documents/Replication/ReplicationNode.cs:47]: Why did you remove the `Url` comparison?

Comment (fitzchak) [reply src/Raven.Client/Documents/Replication/ReplicationNode.cs:47]: Only for external replication, as I understand that it would be null on the beginning and we set this value later on.

Comment (ayende) [reply src/Raven.Client/Documents/Replication/ReplicationNode.cs:47]: Pretty sure that this shouldn't be the case, but for external replication, it uses ConnectionStringName, no? @karmeli87 Can you comment?

Comment (karmeli87) [reply src/Raven.Client/Documents/Replication/ReplicationNode.cs:47]: In external replication we fetch the destination url from the cluster that resides on in the Urls that defined in the connection string, so we can't compare two external replication tasks to be based on the Url.

Comment (ayende) [reply src/Raven.Client/Documents/Replication/ReplicationNode.cs:47]: So how do we detect if two external replication are the same or not?

Comment (fitzchak) [reply src/Raven.Client/Documents/Replication/ReplicationNode.cs:47]: http://issues.hibernatingrhinos.com/issue/RavenDB-9171
